### PR TITLE
Making a subpackage of tomo protocols

### PR DESCRIPTION
**Pull request by dmaluenda**
Tomo protocols are moved to a single subpackage.

Alternatively, we have to consider using a dedicated plugin for xmipp-tomo. This should depends of xmipp's and tomo's plugins, in this way it would be better managed.

### CODE REVIEW

File: xmipp3/protocols/__init__.py [hunk 1]
```diff
@@ -127,14 +126,11 @@
...
-    from .protocol_connected_components import XmippProtConnectedComponents
-    from .protocol_roiIJ import XmippProtRoiIJ
-
-except ImportError:
-    print(
-        'To use a Tomography protocol scipion-em-tomo plugin is required.'
-        ' See https://github.com/scipion-em/scipion-em-tomo for further details')
+    from .tomo import *
+except Exception:
+    from pyworkflow.utils import pluginNotFound
```
Comment (DStrelak) [inline xmipp3/protocols/__init__.py]: Please double check that this will work properly on machines without tomography plugin installed.

Comment (dmaluenda) [reply xmipp3/protocols/__init__.py]: Yes, I've checked it. Do you see any conflicting point?

**Comment by dmaluenda** [comment]
`importFromPlugin` added to the projectZ test in order to be more explicit that tomo's plugin is not found and this is what is failing.

Then, when the tomo's plugin is not found and the tests is run, we get the following error:
```
...

[   FAILED ] TestXmippProtProjectZ.test_top

Traceback (most recent call last):
  File "/home/david/scipionDEVEL/software/lib/python2.7/unittest/case.py", line 329, in run
    testMethod()
  File "/home/david/xmipp-bundle/src/scipion-em-xmipp/xmipp3/tests/test_protocols_xmipp_2d.py", line 1146, in test_top
    prot = self._createProjZ(protImportVols.outputVolumes)
  File "/home/david/xmipp-bundle/src/scipion-em-xmipp/xmipp3/tests/test_protocols_xmipp_2d.py", line 1137, in _createProjZ
    doRaise=True)
  File "/home/david/scipionDEVEL/pyworkflow/utils/utils.py", line 788, in importFromPlugin
    pluginNotFound(plugName, errMsg, doRaise)
  File "/home/david/scipionDEVEL/pyworkflow/utils/utils.py", line 767, in pluginNotFound
    raise Exception("\n\n"+raiseMsg)
Exception: 

 > error when importing from xmipp3: 'module' object has no attribute 'XmippProtSubtomoProject'. Tomography protocols couldn't be imported.
   To use a Tomography protocol scipion-em-tomo plugin is required.
 See https://github.com/scipion-em/scipion-em-tomo for further details
   Called by /home/david/xmipp-bundle/src/scipion-em-xmipp/xmipp3/tests/test_protocols_xmipp_2d.py, line 1137
   Check the plugin manager (Configuration->Plugins in Scipion manager window) 
   or use 'scipion installp --help --checkUpdates' in the command line to check for upgrades,
   it could be a versions compatibility issue.

```

File: xmipp3/protocols/__init__.py (hunk 1 above)
Comment (DStrelak) [reply xmipp3/protocols/__init__.py]: Not this time (but I didn't test the code).

**Comment by EstrellaFG** [comment]
All Xmipp protocols for **tomography** have been **removed** from this plugin to avoid that SPA users have issues with tomography stuff if they don't have installed *scipion-em-tomo* plugin (https://github.com/scipion-em/scipion-em-tomo). 
Tomography protocols are now in this new plugin: https://github.com/I2PC/scipion-em-xmipptomo (which will depend on scipion-em-tomo and scipion-em-xmipp)

File: xmipp3/protocols/protocol_preprocess/__init__.py [hunk 2]
```diff
@@ -24,6 +24,9 @@
 # *
 # **************************************************************************
 
+from .protocol_add_noise import XmippProtAddNoiseVolumes
+from .protocol_add_noise import XmippProtAddNoiseParticles
```
Comment (DStrelak) [inline xmipp3/protocols/protocol_preprocess/__init__.py]: this seems like it might not belong to here (file or PR)

File: xmipp3/wizards.py [hunk 3]
```diff
@@ -29,6 +29,7 @@
 from pyworkflow.em.constants import *
 from pyworkflow.em.wizard import *
 from pyworkflow.em.protocol import ProtImportCoordinates
+from pyworkflow.utils import importFromPlugin
```
Comment (DStrelak) [inline xmipp3/wizards.py]: is this import used?

Comment (EstrellaFG) [reply xmipp3/wizards.py]: https://github.com/I2PC/scipion-em-xmipp/pull/191/commits/5299854b18238d9262a42f3ff3c71477546791c3

File: xmipp3/protocols/protocol_preprocess/__init__.py (hunk 2 above)
Comment (dmaluenda) [reply xmipp3/protocols/protocol_preprocess/__init__.py]: Yes, my fault!

**Comment by dmaluenda** [comment]
Great!

Please, consider to migrate also `mltomo`, `cltomo` and `monotomo`. These are not producing any import issue, but it might make sense to be with the rest of tomo protocols.

For the rest, I agree with this :) Even though, I cannot approve the PR since I created it XD.

**Comment by DStrelak** [comment]
@dmaluenda, don't revert bugs, we're trying to get rid of them!
(just joking :D )

**Pull request merged**